qiskit/synthesis/evolution/qdrift.py

```diff
@@ -0,0 +1,110 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""QDrift Class"""
+
+from typing import List, Union, Optional, Callable, Tuple
+import numpy as np
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.quantum_info.operators import SparsePauliOp, Pauli
+from qiskit.utils import algorithm_globals
+
+from .product_formula import ProductFormula
+
+# pylint: disable=invalid-name
```

Comment:
Can we avoid this global disable? 😛

Reply:
I don't see a reason to have even a local disable in this file - I assume it's complaining about the variable `N`, which would be better being something more descriptive.

Reply:
I renamed it to `num_gates` because that's how this N is referred to in the paper.